### ISSUE TITLE
Selecting markers on the map by drawing an polygon and adding to another zone

### ISSUE THREAD

**Issue by johnGarcia123** [Mapotempo/mapotempo-web#13]
Selecting multiple markers on the map by drawing an polygon and adding to another zone would make easier for high number of destinations instead of one by one.. 
Leaflet seems to not have such as selector.. can be any oportunity for multimarkers selection?

**Comment by frodrigo** [comment]
I would be a nice feature to have. But nobody implement it, yet.

**Comment by johnGarcia123** [comment]
`map.on("boxzoomend", function(e) {
  for (var i = 0; i < markers.length; i++) {
    if (e.boxZoomBounds.contains(markers[i])) {
      console.log(markers[i]);
    }
  }
});
`

something like this piece of code would do the job

**Comment by frodrigo** [comment]
We are thinking on something based on this: https://github.com/Wildhoney/Leaflet.FreeDraw
The hard part is the usage of the selection, not the selection itself.
PR welcome.

**Comment by johnGarcia123** [comment]
What do you mean usage of the selection?

**Comment by frodrigo** [comment]
Apply some action to the selection, like move to other vehicle.

**Comment by johnGarcia123** [comment]
[http://www.zoopla.co.uk/for-sale/map/property/london/?include_retirement_homes=true&include_shared_ownership=true&new_homes=include&q=London&results_sort=newest_listings&search_source=home&pn=1&view_type=map](url)

this looks doing that

**Comment by johnGarcia123** [comment]
[http://www.zoopla.co.uk/for-sale/map/property/london/?include_retirement_homes=true&include_shared_ownership=true&new_homes=include&q=London&results_sort=newest_listings&search_source=home&pn=1&view_type=map](url)

this looks doing that

**Comment by frodrigo** [comment]
Yes it can be done, one by one. But on numerous points it will be slow.
It should be done by using the mass move API entry point: http://swagger.mapotempo.com/#!/plannings/moveVisits

**Comment by johnGarcia123** [comment]
`https://freedraw.herokuapp.com/` 
could this also be implemented on zonings?

**Comment by frodrigo** [comment]
Yes, it's a nice featured plugin.
We aim to improve the zoning draw part one day.

**Comment by johnGarcia123** [comment]
`var map = new L.Map('leaflet', {
    'center': [0, 0],
    'zoom': 0
});

var markers = new L.LayerGroup().addTo(map);

for (var i = 0; i < 300; i++) {
    var marker = new L.Marker([
        (Math.random() * (90 - -90) + -90).toFixed(5) * 1,
        (Math.random() * (180 - -180) + -180).toFixed(5) * 1
    ]).addTo(markers);
}

new L.Control.Draw({
    draw: {
        marker   : false,
        polygon  : true,
        polyline : false,
        rectangle: true,
        circle   : {
            metric: 'metric'
        }
    },
    edit: false
}).addTo(map);

L.Polygon.include({
    contains: function (latLng) {
        return turf.inside(new L.Marker(latLng).toGeoJSON(), this.toGeoJSON());
    } 
});

L.Rectangle.include({
    contains: function (latLng) {
        return this.getBounds().contains(latLng);
    }
});

L.Circle.include({
    contains: function (latLng) {
        return this.getLatLng().distanceTo(latLng) < this.getRadius();
    }
});

map.on(L.Draw.Event.CREATED, function (e) {
    markers.eachLayer(function (marker) {
        if (!e.layer.contains(marker.getLatLng())) {
            marker.remove();
        }
    });
});
`

`<!DOCTYPE html>
<html>
  <head>
    <title>Leaflet 1.0.3</title>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link type="text/css" rel="stylesheet" href="//unpkg.com/leaflet@1.0.3/dist/leaflet.css" />
    <link type="text/css" rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/leaflet.draw/0.4.9/leaflet.draw.css" />
  </head>
  <body>
    <div id="leaflet"></div>
    <script type="application/javascript" src="//unpkg.com/leaflet@1.0.3/dist/leaflet.js"></script>
    <script type="application/javascript" src="//cdnjs.cloudflare.com/ajax/libs/leaflet.draw/0.4.9/leaflet.draw.js"></script>
    <script type="application/javascript" src="//unpkg.com/@turf/turf@latest/turf.min.js"></script>
  </body>
</html>
`

It works

**Comment by johnGarcia123** [comment]
ive already implemented kinda selector.. but what kinda of types should i select over the map.. there is not marker , is it an L.Point?
